### DISCUSSION TITLE
non-www to www as default domain

### DISCUSSION THREAD

**Discussion by golivews** [NginxProxyManager/nginx-proxy-manager#1683] [Q&A]
How can I add Custom Nginx Configuration to automatically redirect non-www to www as default domain (https)?
Thanks in advance!

**Comment by chaptergy** [comment]
<sup>[Answered by @ch4ox](https://github.com/jc21/nginx-proxy-manager/issues/1684#issuecomment-1001579308)</sup>

That's what "Redirection Hosts" are for 😉

Example: You want to have everything redirect to www.example.org:
- Set up a "Proxy Host" which only covers www.example.org
- Set up a "Redirection Host" with HTTP code 301 which covers example.org and *.example.org and forwards to https://www.example.org (make sure to tick "Preserve Path")

Example: You want to have everything redirect to example.org:
- Set up a "Proxy Host" which only covers example.org
- Set up a "Redirection Host" with HTTP code 301 which covers *.example.org and forwards to https://example.org (make sure to tick "Preserve Path")

The important thing to know is that "Proxy Hosts" always have priority, that's why you can have a "Redirection Host" with *.example.org and it won't cover www.example.org if there's already a Proxy Host for that. 

BTW: If you tick "Force SSL" on your "Proxy Host" there will be a redirect to HTTPS as well.

**Reply by golivews** [reply]
Thank you @chaptergy

**Reply by jkavenaugh** [reply]
Does this still work? I'm not having any luck.
If I Force SSL I get net::ERR_CERT_COMMON_NAME_INVALID it doesn't change to www.
If I untick Force SSL I get ERR_SSL_UNRECOGNIZED_NAME_ALERT
